### ISSUE TITLE
Missing selection

### ISSUE THREAD

**Issue by kaihendry** [cdown/clipmenu#87]
Using Archlinx's 5.5.0-1 enabled by

```
readlink ~/.config/systemd/user/default.target.wants/clipmenud.service
/usr/lib/systemd/user/clipmenud.service
```

```
[hendry@t480s ~]$ clipmenu
FATAL: db.users.findOne({'emails.address': 'kai.hendry@gmail.com'}) not in cache (/run/user/1000/clipmenu.5.hendry/4256356820 61 missing)
Please report the following debug information:

 1000 /run/user/1000/clipmenu.5.hendry/line_cache_clipboard
 1000 /run/user/1000/clipmenu.5.hendry/line_cache_primary
 2000 total
/run/user/1000/clipmenu.5.hendry/line_cache_primary:110:1537855934453509012 db.users.findOne({'emails.address': 'kai.hendry@gmail.com'})
stat: cannot stat '/run/user/1000/clipmenu.5.hendry/4256356820 61': No such file or directory
```

Perhaps if it can't find the record, it could fall back to whatever is in the line_cache?

I am puzzled, why I see these messages in my logs:

```
[hendry@t480s clipmenu.5.hendry]$ journalctl -b | grep 4256356820
Sep 26 17:40:43 t480s clipmenud[811]: rm: cannot remove '/run/user/1000/clipmenu.5.hendry//4256356820 61': No such file or directory
Sep 27 15:06:17 t480s clipmenud[26467]: rm: cannot remove '/run/user/1000/clipmenu.5.hendry//4256356820 61': No such file or directory
Sep 27 15:06:29 t480s clipmenud[26467]: rm: cannot remove '/run/user/1000/clipmenu.5.hendry//4256356820 61': No such file or directory
```
A partial cliip ? https://github.com/cdown/clipmenu/blob/develop/clipmenud#L159
Why is it removing again?

**Comment by cdown** [comment]
> Perhaps if it can't find the record, it could fall back to whatever is in the line_cache?

I'm somewhat against silent failure.

To debug this reasonably, `CM_DEBUG` needs to be on at the time the issue starts (having it right now won't help).

To be honest I should probably just make some of the messages that are currently gated behind `CM_DEBUG` on by default.

**Comment by qstrejo** [comment]
Do you by any chance have a non-GNU date installed? I ran into the same issue when I had heirloom date in my PATH.

**Comment by kaihendry** [comment]
Got the GNU:

```
[hendry@t480s ~]$ date --version
date (GNU coreutils) 8.30
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by David MacKenzie.
```

**Comment by asheshambasta** [comment]
I can confirm that AFAICS, this starts happening after I've suspended the system and resumed it back. 
I see 
```
Jul 09 09:18:46 quasar-nixos-tr xsession[1967]: rm: cannot remove '/run/user/1000/clipmenu.5.ashesh//903472502 36': No such file or directory
```

What I don't understand is what is up with the `//` and the spaces in the filename? Is that expected?

**Comment by cdown** [comment]
> What I don't understand is what is up with the // and the spaces in the filename? Is that expected?

Yes, both of these are expected (`//` is equivalent to `/`, and the spaces are totally normal).

I can't think of any reason why suspend should affect this, since we don't do anything based on (for example) realtime, or system state.

I'm going to start outputting more information to stdout without CM_DEBUG on, which hopefully will help debug for people which are having this issue :-)

**Comment by cdown** [comment]
Ok, I found one case where this could happen. This can happen if you copy the same thing (or at least the same first line and number of lines) twice, which results in us overwriting the original. So far so good, this is an intentional design decision. However, if we have `CM_MAX_CLIPS` set (which it is by default) and a duplicated clip falls off the edge, it will result in deleting the file even if we still have an entry for it in the line cache.

As such, we should make our checks more robust before deleting the backing clip file.

**Comment by cdown** [comment]
Please try with latest `develop` branch and a fresh clipmenu cache (it won't fix already removed entries), it should fix this issue :-)

**Comment by cdown** [comment]
Specifically, 85aaec162665fac3f85b82d5f3fdbb826b1f4bc3 should fix it, and a495bcc7a4ab125182a661c5808364f66938a87c amortises the performance penalty of the new check

**Comment by cdown** [comment]
6.0.0 has been released with the fixes in. Please let me know if you still experience this with that version. Thanks for the report! :-)
